datafusion/sql/src/unparser/expr.rs

```diff
@@ -305,10 +352,10 @@ impl Unparser<'_> {
                 })
             }
             Expr::Sort(Sort {
-                expr,
+                expr: _,
                 asc: _,
                 nulls_first: _,
-            }) => self.expr_to_sql(expr),
+            }) => internal_err!("Sort expression should be handled by expr_to_unparsed"),
```

Comment:
Since this is a public API I don't think this should be an internal_err as it isn't a bug in datafusion (it could be a bug in the input that was passed in)-- perhaps we can make it a `plan_err`?

Reply:
Ohh, great. Now I know better when to use these types of error.